### ISSUE TITLE
Inconsistent parameter names

### ISSUE THREAD

**Issue by msh321** [swagger-api/swagger-codegen#1291]
I have the following Swagger Specification:
```yaml
swagger: '2.0'
info:
  description: TODO
  version: 0.0.2
  title: TODO
  termsOfService: TODO
host: 10.11.12.13:8080
basePath: /v7
schemes:
  - http
  - https
paths:
# ==============================================================================
# == Asynchronous REST operations (create, delete, modify, recreate service)
# ==============================================================================
  /abc/serv:
    get:
      tags:
        - serv
      summary: TODO
      description: 'TODO'
      operationId: getServ
      consumes:
        - application/json
        - application/xml
      produces:
        - application/json
        - application/xml
      parameters:
        - in: header
          name: X-Token
          description: TODO
          required: true
          type: string
        - in: header
          name: Callback
          description: TODO
          required: true
          type: string
        - in: query
          name: id
          description: TODO
          required: true
          type: boolean
      responses:
        '200':
          description: Request was accepted
```
Here is the documentation from the Swagger Editor:

![img1](https://cloud.githubusercontent.com/assets/1345452/10070580/adfa37a0-62b3-11e5-8b88-a98c4770cc58.png)

And this is what I get when using `codegen` (language:`dynamic-html`):

![img2](https://cloud.githubusercontent.com/assets/1345452/10070587/b40a10c0-62b3-11e5-9dfd-eec2d97ab1fb.png)

Notice that `X-Token` has been renamed to `xToken` and `Callback` to `callback`.

I need to align with an existing API, so I cannot change the parameter names.

BTW: The use of the `Callback` parameter leads to another problem; namely that the generated Java files cannot compile (because `callback` is used for other purposes in the auto-generated code). I just made a posting on the forum to see if there is a way to work around this "name collision issue".

**Comment by wing328** [comment]
For the parameter naming issue (e.g. callback), please try with the `localVariablePrefix` in the config option:
```
swagger-codegen|master⚡ ⇒ java -jar modules/swagger-codegen-cli/target/swagger-codegen-cli.jar config-help -l java 

CONFIG OPTIONS
	modelPackage
	    package for generated models

	apiPackage
	    package for generated api classes

	invokerPackage
	    root package for generated code

	groupId
	    groupId in generated pom.xml

	artifactId
	    artifactId in generated pom.xml

	artifactVersion
	    artifact version in generated pom.xml

	sourceFolder
	    source folder for generated code

	localVariablePrefix
	    prefix for generated code members and local variables

	serializableModel
	    boolean - toggle "implements Serializable" for generated models

	library
	    library template (sub-template) to use:
	    <default> - HTTP client: Jersey client 1.18. JSON processing: Jackson 2.4.2
	    jersey2 - HTTP client: Jersey client 2.6
	    okhttp-gson - HTTP client: OkHttp 2.4.0. JSON processing: Gson 2.3.1
```

**Comment by wing328** [comment]
For the parameter naming issue in `dynamic-html`, it should not be a difficult fix so I should be able to submit a PR by next Monday.

**Comment by msh321** [comment]
Thanks a lot for the help and for the responsiveness.

I have tried out the `localVariablePrefix` as you suggested, and it seems to do the trick in my specific case for the naming collisions.

However, in the more _general_ case, shouldn't it be the API variables that should get the prefix?  I mean, if I -- for instance -- need a HTTP header param named "Import", then in the auto-generated Java code, the name would turn into "import" which collides with Java's reserved keywords.  :smirk_cat:

**Comment by wing328** [comment]
For reserved keywords, it should be appended with an _ at the beginning (e.g. _import). Please give it a try and let me know if that's no the case.

**Comment by msh321** [comment]
Your absolutely correct. I tried to add a HTTP header param named `Import` and in the generated Java code, the corresponding variable was converted to `_import`.  So please forget my concern.  :+1:

As for the generation of "dynamic HTML" I believe that I have found another issue related to the request body.  If I have an API method with a request body that refers to a definition, then there is no hyperlink from the body to the documentation of the body.

The following example illustrates this:

```yaml
swagger: '2.0'
info:
  description: TODO
  version: 0.0.1
  title: API Spec
  termsOfService: TODO
basePath: /v1
schemes:
  - http
  - https
paths:
  /aaa:
    post:
      tags:
        - AAA
      summary: TODO
      description: TODO
      operationId: aaa
      consumes:
        - application/json
        - application/xml
      produces:
        - application/json
        - application/xml
      parameters:
        - in: header
          name: X-Token
          description: TODO
          required: true
          type: string
        - in: header
          name: Import
          description: TODO
          required: true
          type: string
        - in: body
          name: body
          description: TODO
          required: true
          schema:
            $ref: '#/definitions/AAA'
      responses:
        '200':
          description: Authentication was successful
        '405':
          description: Invalid data
definitions:
  AAA:
    allOf:
      - $ref: '#/definitions/BBB'
      - type: object
        properties:
          ttt:
            type: string
  BBB:
    allOf:
      - $ref: '#/definitions/CCC'
      - type: object
        properties:
          vvv:
            type: string
  CCC:
    type: object
    properties:
      uuu:
        type: string
      ppp:
        type: string
```

If I generate "dynamic HTML" from this, then the body object is not clickable.

**Comment by wing328** [comment]
@msh321 I've fixed the parameter name in #1310. Please do a test and let me know if you still experience the issue.

**Comment by msh321** [comment]
Hmmmm -- I just updated be rebuilt.  But my HTTP header parameter names are still wrong.

So, `Callback` is still renamed to `callback` and `X-Token` is renamed to `xToken`.

**Comment by wing328** [comment]
@msh321 The fix has not been merged into master. You need to clone https://github.com/wing328/swagger-codegen and checkout fix_dynamic_html_parameter_name in order to test the fix.

**Comment by msh321** [comment]
@wing328 I have now tried out the `fix_dynamic_html_parameter_name` branch, and the fix work. My HTTP header params are now correct in the generated docs.  Thanks a lot!  :smiley: 

I have discovered a few other issue (I will create a issue for them):
1. If a REST method has a "body", then the body is not clickable.
1. If I click on a response object, and the response object contains other object, then these other objects are not clickable.
1. If a REST method has multiple reponses (e.g. 200, 401, 500, ...) then only a single response seems to get rendered.
1. If the Swagger Specification contains a "global response" section, then I cannot see it anywhere in the generated HTML

Anyway -- thanks a lot for the excellent work...
